### PR TITLE
Store Pepys version and build date (if available) in the Changes.reason field when running import

### LINKED ISSUE #949: Report Build version in extraction log
When we import a file, we log it in the `Changes` log.

In this report, we should include the version and build-date of the Pepys instance in use.  This will allow us to reconstruct the import operation, if necessary.

[Request from CF]

### PR DESCRIPTION
**Pull request by robintw**
## 🧰 Issue
Fixes #949 

## 🚀 Overview: 
Extends the import process to store the version of Pepys and the build date (if available - it's only available for releases) in the `reason` field of the `Changes` entry created for the import. The `reason` will now look like:

```
Importing 'rep_test1.rep' using Pepys 0.0.25, built on 2021-01-01 17:00:00
```

## 🤔 Reason: 
Gives a better audit trail to determine which files were imported with which versions.

## 🔨Work carried out:

- [x] Altered FileProcessor to change reason
- [x] Added tests
- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have chosen an appropriate label for the PR, adding `interactive_review` if reviewers will need to see UI
- [x] I have extended/updated the documentation in `\docs` folder
- [x] Any database content changes (Create, Edit, Delete) are recorded in the Log/Changes tables
- [x] Any database schema changes are implemented via `alembic revision` [transitions](https://pepys-import.readthedocs.io/en/latest/database_migration.html#how-to-use-it-for-developers)
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.

### CODE REVIEW

File: tests/test_importers.py [hunk 1]
```diff
@@ -1033,5 +1035,38 @@ def test_skip_file(self, resolver_prompt, menu_prompt, file_processor_prompt):
             assert len(self.store.session.query(self.store.db_classes.Comment).all()) == comments
 
 
+def test_changes_table_entry_contains_version():
```
Comment (mew-nsc) [inline tests/test_importers.py:1038]: Test for version looks good Robin. Would it be possible to have another test to check `__build_timestamp__` being applied?

**Review by mew-nsc — APPROVED** [review]

File: tests/test_importers.py (hunk 1 above)
Comment (robintw) [reply tests/test_importers.py:1038]: Unfortunately that would be fairly difficult - as the build timestamp is only set when we do a full release. That means that when we're running the tests on the development version, `__build_timestamp__` will always be None.

Comment (IanMayo) [reply tests/test_importers.py:1038]: Aah, as it happens I'm going to make a new release today -  so I can offer a manual test to verify it.

**Review by IanMayo — APPROVED** [review]

File: tests/test_importers.py (hunk 1 above)
Comment (mew-nsc) [reply tests/test_importers.py:1038]: No problem Robin. If you're happy to test as part of your release Ian, that would be great.

**Pull request merged**